test/AutoDiff/forward_mode_runtime.swift

```diff
@@ -0,0 +1,308 @@
+      return (f(x), { (dself, dx) in dself.base * dx })
+    }
+    final func vjpf(_ x: Float) -> (Float, (Float) -> (TangentVector, Float)) {
+      let base = self.base
+      return (f(x), { v in
+        (TangentVector(base: v * x, _nontrivial: []), base * v)
+      })
+    }
+  }
+
+  class SubOverride : Super {
+    @differentiable(wrt: (self, x))
+    override func f(_ x: Float) -> Float {
+      return 3 * x
+    }
+  }
+
+  class SubOverrideCustomDerivatives : Super {
+    @differentiable(wrt: (self, x))
+    @differentiable(wrt: x, jvp: jvpf2, vjp: vjpf2)
+    override func f(_ x: Float) -> Float {
+      return 3 * x
+    }
+    final func jvpf2(_ x: Float) -> (Float, (Float) -> Float) {
+      return (f(x), { v in 3 * v })
+    }
+    final func vjpf2(_ x: Float) -> (Float, (Float) -> Float) {
+      return (f(x), { v in 3 * v })
+    }
+  }
+
+  // TODO(TF-654): Uncomment when differentiation supports class initializers.
+  // let v = Super.TangentVector(base: 100, _nontrivial: [])
+  // expectEqual(100, pullback(at: 1337) { x in Super(base: x) }(v))
+  // expectEqual(100, pullback(at: 1337) { x in SubOverride(base: x) }(v))
+  // expectEqual(100, pullback(at: 1337) { x in SubOverrideCustomDerivatives(base: x) }(v))
+  
+
+  // `valueWithGradient` is not used because nested tuples cannot be compared
+  // with `expectEqual`.
+  func classGradient(_ c: Super) -> (Super.TangentVector, Float) {
+    return gradient(at: c, 10) { c, x in c.f(x) }
+  }
+
+  // `valueWithDerivative` is not used because the derivative requires `Super`
+  // to conform to `FloatingPoint`.
+  func classDifferential(
+    _ c: Super
+  ) -> (Float, (Super.TangentVector, Float) -> Float) {
+    return valueWithDifferential(at: c, 10) { (c: Super, x: Float) in c.f(x) }
+  }
+
+  let (y1, diff1) = classDifferential(Super(base: 5))
+  expectEqual(y1, 50)
+  expectEqual(diff1(Super.TangentVector(base: 1, _nontrivial: []), 1), 1)
+  let (y2, diff2) = classDifferential(SubOverride(base: 5))
+  expectEqual(y2, 30)
+  expectEqual(diff2(SubOverride.TangentVector(base: 1, _nontrivial: []), 1), 3)
+  let (y3, diff3) = classDifferential(SubOverrideCustomDerivatives(base: 5))
+  expectEqual(y3, 30)
+  expectEqual(diff3(SubOverrideCustomDerivatives
+                      .TangentVector(base: 1, _nontrivial: []), 1), 3)
+  expectEqual((Super.TangentVector(base: 10, _nontrivial: []), 2),
+              classGradient(Super(base: 2)))
+  expectEqual((Super.TangentVector(base: 0, _nontrivial: []), 3),
+              classGradient(SubOverride(base: 2)))
+  expectEqual((Super.TangentVector(base: 0, _nontrivial: []), 3),
+              classGradient(SubOverrideCustomDerivatives(base: 2)))
+}
+
+//===----------------------------------------------------------------------===//
+// Protocols
+//===----------------------------------------------------------------------===//
+// TODO: add more protocol tests.
+protocol DiffReq : Differentiable {
+  func foo(x: Float) -> Float
```

Comment:
The point of testing protocols is testing requirements that are `@differentiable`. Please add `@differentiable(wrt: x)`.